src/plugins/data/common/data_views/data_views/data_view.ts

```diff
@@ -391,24 +425,38 @@ export class DataView implements IDataView {
     );
   }
 
-  /**
-   * Converts a DataView to a serializable Dataset object suitable for storage in Redux
-   * Maps dataSource to dataSourceRef and includes only essential properties
-   * TODO: This should be more complete
-   */
-  public toDataset(): Dataset {
+  public async toDataset(): Promise<Dataset> {
+    const defaultType = DEFAULT_DATA.SET_TYPES.INDEX_PATTERN;
+    const dataSourceReference = this.dataSourceRef || (this as any).dataSource;
+
+    let dataSource;
+    if (dataSourceReference?.id) {
+      try {
+        const dataSourceSavedObject = await this.savedObjectsClient.get(
+          dataSourceReference.type,
+          dataSourceReference.id
+        );
+        const attributes = dataSourceSavedObject.attributes as any;
+        dataSource = {
+          id: dataSourceReference.id,
+          title: attributes.title || dataSourceReference.id,
+          type: attributes.dataSourceEngineType || dataSourceReference.type || 'OpenSearch',
+        };
+      } catch (error) {
+        dataSource = {
+          id: dataSourceReference.id,
+          title: dataSourceReference.name || dataSourceReference.title || dataSourceReference.id,
```

Comment:
what's the difference between name and title?

Reply:
I believe name was for legacy (ie index pattern matcher with the * at the end). Title is a new alias we can have for the dataset to help users identify the dataset

Or it's the other way around, I forget